reader_concurrency_semaphore.cc

```diff
@@ -155,7 +155,7 @@ class reader_permit::impl
     void on_timeout() {
         _as.request_abort_ex(std::make_exception_ptr(timed_out_error{}));
 
-        if (_state == state::waiting_for_admission || _state == state::waiting_for_memory) {
+        if (_state == state::waiting_for_admission || _state == state::waiting_for_memory || _state == state::waiting_for_execution) {
```

Comment:
I wonder if we could have implemented waiting_for_memory by pushing it to the front of the admission queue.

Reply:
I considered it, but it would have been very awkward (or even impossible) with the previous container (`expiring_fifo`). It is possible to do now, so maybe I should reconsider it. On the other hand intrusive lists are very cheap (but not free).